src/algorithms/reco/JetReconstruction.cc

```diff
@@ -46,34 +46,39 @@ namespace eicrecon {
       m_log->error(" Unknown area type \"{}\" specified!", m_cfg.areaType);
       throw JException(out.what());
     }
-  }
+  }  // end 'init(std::shared_ptr<spdlog::logger>)'
 
 
 
-  std::unique_ptr<edm4eic::ReconstructedParticleCollection> JetReconstruction::process(
-    const std::vector<const edm4hep::LorentzVectorE*> momenta) {
+  template <typename T> std::unique_ptr<edm4eic::ReconstructedParticleCollection> JetReconstruction::process(const T& input_collection) {
```

Comment:
There is no way for a compiler to figure out which instantiations it needs to emit in this translation unit, so you'll need to add these two lines below:
```c++
template std::unique_ptr<edm4eic::ReconstructedParticleCollection> JetReconstruction::process(const edm4hep::MCParticleCollection& input_collection);
template std::unique_ptr<edm4eic::ReconstructedParticleCollection> JetReconstruction::process(const edm4eic::ReconstructedParticleCollection& input_collection);
```